cmd/kubelet/app/server.go

```diff
@@ -124,25 +127,65 @@ is checked every 20 seconds (also configurable with a flag).
 
 HTTP server: The kubelet can also listen for HTTP and respond to a simple API
 (underspec'd currently) to submit a new manifest.`,
+		// The Kubelet has special flag parsing requirements to enforce flag precedence rules,
+		// so we do all our parsing manually in Run, below.
+		// DisableFlagParsing=true provides the full set of flags passed to the kubelet in the
+		// `args` arg to Run, without Cobra's interference.
+		DisableFlagParsing: true,
 		Run: func(cmd *cobra.Command, args []string) {
-			// short-circuit on verflag
-			verflag.PrintAndExitIfRequested()
-
-			// TODO(mtaufen): won't need this this once dynamic config is GA
-			// set feature gates so we can check if dynamic config is enabled
-			if err := utilfeature.DefaultFeatureGate.SetFromMap(kubeletConfiguration.FeatureGates); err != nil {
+			// initial flag parse, since we disable cobra's flag parsing
+			cmd.Flags().Parse(args)
+			// set feature gates from initial flags-based config
+			if err := utilfeature.DefaultFeatureGate.SetFromMap(kubeletConfig.FeatureGates); err != nil {
 				glog.Fatal(err)
 			}
-			// validate the initial KubeletFlags, to make sure the dynamic-config-related flags aren't used unless the feature gate is on
+			// log flags
+			cmd.Flags().VisitAll(func(flag *pflag.Flag) {
+				glog.V(2).Infof("FLAG: --%s=%q", flag.Name, flag.Value)
```

Comment:
do all the flag values print with %q as expected? would it be better to just log the args?

Reply:
logging args now